File: src/main/java/ch.bfh.eadj.bookstore/entity/Book.java
package ch.bfh.eadj.bookstore.entity;

import java.math.BigDecimal;
import javax.persistence.Entity;
import javax.persistence.Id;

/**
 * Created by philippewanner on 14.11.16.
 */

@Entity
public class Book {
    
    @Id
    private String isbn;
    
    private String title;
    
    private String authors;
    
    private String publisher;
    
    private Integer publicationYear;
    
    private BookBinding binding;
    
    private Integer numberOfPages;
    
    private BigDecimal price;
    
    public enum BookBinding {
        HARD_COVER, SOFT_COVER
    }
}

File: client/app/core/book.ts
export class Book {

    constructor(public isbn: string,
                public title: string = "",
                public authors: string = "",
                public publisher: string = "",
                public publicationYear: number = 1888,
                public binding: BookBinding = BookBinding.UNKNOWN,
                public numberOfPages: number = 0,
                public price: number=0) {
    }

    public containsKeywords(keywords: string[]): boolean{
        let bookString: string =this.isbn + " " + this.title + " " + this.authors + " " + this.publisher;

        let contains: boolean=true;

        for (let word of keywords) {
            contains = contains && (bookString.includes(word));
        }

        return contains;
    }

}

export enum BookBinding {
    HARDCOVER, PAPERBACK, EBOOK, UNKNOWN
}